### PR TITLE
Cyphers/bnorm back

### PR DESCRIPTION
**Pull request by diyessi**
Provide BatchNorm constructors with normal arg order (old versions deprecated)
Fix BatchNorm reference implementations.
Add tests with crafted numbers.
@jbobba the autodiff test gets a compilation error on CPU. Disabled in manifest.

### CODE REVIEW

File: src/ngraph/op/batch_norm.hpp [hunk 1]
```diff
@@ -27,9 +27,20 @@ namespace ngraph
...
+            // \param input Must have rank >= 2, [., C, ...]
+            // \param gamma gamma scaling for normalized value. [C]
+            // \param beta bias added to the scaled normalized value [C]
+            // \param epsilon Avoids divsion by 0 if input has 0 variance
+            BatchNormTraining(std::shared_ptr<Node> input,
+                              std::shared_ptr<Node> gamma,
+                              std::shared_ptr<Node> beta,
+                              double epsilon);
+
+            // DEPRECATED
```
Comment (rkimballn1) [inline src/ngraph/op/batch_norm.hpp]: For future reference, there is a \deprecated doxygen tag that should be used.

File: src/ngraph/serializer.cpp [hunk 2]
```diff
@@ -531,21 +531,21 @@ static shared_ptr<ngraph::Function>
             case OP_TYPEID::BatchNormTraining:
             {
                 auto epsilon = node_js.at("eps").get<double>();
-                node = make_shared<op::BatchNormTraining>(epsilon, args[0], args[1], args[2]);
+                node = make_shared<op::BatchNormTraining>(args[2], args[0], args[1], epsilon);
```
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/serializer.cpp]: Or you could assign appropriately named temporaries for arg[0], arg[1] which would hopefully make this less prone to error.

[the remark this answers is not in the document]
Comment (diyessi) [reply src/ngraph/serializer.cpp]: Added a comment.
